astropy/stats/tests/test_funcs.py

```diff
@@ -440,6 +442,42 @@ def test_mad_std():
         data = normal(5, 2, size=(100, 100))
         assert_allclose(funcs.mad_std(data), 2.0, rtol=0.05)
 
+def test_mad_std_scalar_return():
+    with NumpyRNGContext(12345):
+        data = normal(5, 2, size=(10, 10))
+        # make a masked array with no masked points
+        data = np.ma.masked_where(np.isnan(data), data)
+        rslt = funcs.mad_std(data)
+        # want a scalar result, NOT a masked array
+        assert np.isscalar(rslt)
+
+        data[5,5] = np.nan
+        rslt = funcs.mad_std(data, ignore_nan=True)
+        assert np.isscalar(rslt)
+        with catch_warnings() as warns:
+            rslt = funcs.mad_std(data)
+            if NUMPY_GE_1_13:
+                # for recent numpy versions, numpy masked arrays appear to
```

Comment:
Good point, I noticed that the value changed.  In the below, data is a regular old array, `data_m` is masked

```
In [28]: data_m = np.ma.array([1,2,3,4,5.])

In [29]: data_m[2] = np.nan

In [30]: mad_std(data, ignore_nan=True)
Out[30]: 2.223903327758403

In [31]: mad_std(data_m, ignore_nan=True)
Out[31]: 2.223903327758403

In [32]: mad_std(data_m)
Out[32]: 2.9652044370112041
```